### PR TITLE
fix(provisioning-config): multiple bug fixes

### PR DESCRIPTION
**Pull request by JanZachmann**
- fixed bug when awaiting config changes
- fixed x509 config without EST

### CODE REVIEW

File: src/twin/mod.rs [hunk 1]
```diff
@@ -505,8 +507,8 @@ impl Twin {
                         Some(request) = rx_web_service.recv() => {
                             twin.handle_webservice_request(request).await?
                         },
-                        _ = twin.feature::<ProvisioningConfig>()?.next() => {
-                            twin.feature_mut::<ProvisioningConfig>()?.refresh().await?
+                        _ =  Self::notify_some_interval(&mut refresh_prov_certs_interval) => {
```
Comment (empwilli) [inline src/twin/mod.rs:510]: Nit: remove space before Self

File: src/twin/provisioning_config.rs [hunk 2]
```diff
@@ -18,11 +18,22 @@ macro_rules! identity_config_file_path {
...
+                env::var("EST_CERT_FILE_PATH").unwrap_or(EST_CERT_FILE_PATH_DEFAULT.to_string())
+            }
+            false => {
+                const DEVICE_CERT_FILE_PATH_DEFAULT: &'static str =
+                    "/mnt/cert/priv/device_id_cert.pem";
+                env::var("DEVICE_CERT_FILE_PATH")
+                    .unwrap_or(DEVICE_CERT_FILE_PATH_DEFAULT.to_string())
+            }
+        }
+    };
```
Comment (empwilli) [inline src/twin/provisioning_config.rs:39]: We only use this macro once, don't we? Maybe we could move the glob part into the macro, and change it to a function/struct that impls IntoIter<Type=Path>?

File: src/twin/provisioning_config.rs [hunk 3]
```diff
@@ -164,20 +176,20 @@ impl ProvisioningConfig {
...
         let Some(_) = rx.borrow_mut().recv().await else {
             debug!("provisioning_config watcher: sender dropped");
-            return future::pending().right_future();
+            return future::pending().await;
         };
 
         info!("provisioning_config watcher: config changed");
 
-        future::ready(()).left_future()
+        future::ready(()).await
```
Comment (empwilli) [inline src/twin/provisioning_config.rs]: Not quite sure whether I understand what's happening here. Is future::pending().await really what we want? According to the docs this future will never resolve? https://doc.rust-lang.org/std/future/fn.pending.html Doesn't this mean, that the function will never return in the error cases? Conversely, won't future::ready(()) not return immediately? Do we need it then, after all? Wouldn't it be sufficient to use return only in any cases?

**Review by empwilli — CHANGES_REQUESTED** [review]

File: src/twin/provisioning_config.rs (hunk 3 above)
Comment (JanZachmann) [reply src/twin/provisioning_config.rs]: This is not the head of branch you reviewed here. The code you commented is not existing anymore.

Comment (empwilli) [reply src/twin/provisioning_config.rs]: Ack, my bad :)

**Review by JoergZeidler — APPROVED** [review]

**Review by empwilli — APPROVED** [review]

**Pull request merged**